security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/endsession/request/DefaultAuthorizationServerResolver.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.oauth2.endpoint.endsession.request;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import jakarta.inject.Singleton;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * {@link io.micronaut.context.annotation.DefaultImplementation} of {@link AuthorizationServerResolver}.
+ * Based on substrings of the issuer url it returns an {@link AuthorizationServer}.
+ * @author Sergio del Amo
+ * @since 4.12.0
+ */
+@Internal
+@Singleton
+class DefaultAuthorizationServerResolver implements AuthorizationServerResolver {
```

Comment:
can be `final`

Reply:
resolved via https://github.com/micronaut-projects/micronaut-security/pull/1873/commits/715713f3c729e230db09edd89de4d3afc5bd764b